lib/solidus_six_saferpay/address_name_extractor.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module SolidusSixSaferpay
+  class AddressNameExtractor
+    class UnsafeNameExtractionError < StandardError
+    end
+
+    attr_reader :address_name
+
+    def initialize(address)
+      @address_name = extract_names(address)
```

Comment:
IMO the naming is not wrong since it does in fact extract multiple names (first- and lastname), but I totally agree that it is confusing. I changed the name to `extract_first_and_last_name`.